### PR TITLE
Feature/admin user management

### PR DESCRIPTION
**Pull request by o-zaengle**
This adds admin level CRUD for users. 
Allowing creation, deletion, updates, reading, and indexing for users.

Allows a user to be assigned a `current organization`;

### CODE REVIEW

**Comment by jesseschutt** [comment]
@o-zaengle - Can you please rebase this onto develop with the new route naming conventions?

**Comment by o-zaengle** [comment]
@jesseschutt Rebase onto develop complete.

File: Modules/User/Http/Handlers/Admin/AssignUserToOrganizationHandler.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    protected $repository;
+    protected $organizationRepository;
+
+    public function __construct(UserRepository $repository, OrganizationRepository $organizationRepository)
+    {
+        $this->repository = $repository;
+        $this->organizationRepository = $organizationRepository;
+    }
+
+    public function listenTo(Request $request)
```
Comment (jesseschutt) [inline Modules/User/Http/Handlers/Admin/AssignUserToOrganizationHandler.php]: Can we please add the missing docblocks?

File: Modules/User/Tests/Unit/Repositories/UserRepositoryTest.php [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+            'first_name' => 'John',
+            'last_name'  => 'Doe',
+        ]);
+    }
+
+    /** @test */
+    public function it_encrypts_password_when_saving_a_user()
+    {
+        $created = $this->repository->create([
+            'email'      => 'JOhnDoE@eXaMplE.com',
```
Comment (jesseschutt) [inline Modules/User/Tests/Unit/Repositories/UserRepositoryTest.php]: This is a good usage of an implicit test @o-zaengle.

**Pull request merged**